resources/codegen_inputs/stat.rs

```diff
@@ -19,12 +22,15 @@ table Stat {
     #[read_offset_with($design_axis_count)]
     design_axes_offset: Offset32<[AxisRecord]>,
     /// The number of axis value tables.
+    #[compile(array_len($offset_to_axis_value_offsets))]
     axis_value_count: u16,
     /// Offset in bytes from the beginning of the STAT table to the
     /// start of the design axes value offsets array. If axisValueCount
     /// is zero, set to zero; if axisValueCount is greater than zero,
     /// must be greater than zero.
     #[read_offset_with($axis_value_count)]
+    #[compile_type(OffsetMarker<Vec<OffsetMarker<AxisValue>>, WIDTH_32>)]
+    #[to_owned(convert_axis_value_offsets(obj.offset_to_axis_values()))]
```

Comment:
Holy leaking Rust into schema. How might we avoid having to do this, have the codegen be able to figure it out from less Rust-specific information?

(maybe we should just add that to the appropriate issue)

Reply:
This is 100% specific to the architecture of this crate, and is also a design choice. To walk through how we get here:

- this field is an *offset* to an *array of offsets*
- this means that there are two levels of indirection
- we have a mechanism for indicating arguments for resolving an offset, but if there are multiple layers of offsets in a given field this mechanism becomes ambiguous: do the arguments apply to the first offset or the second?
- This is easy to solve with a shim table, which provides us a simple way to separate the layers, and specify attributes for each one
- and that generates reasonable code for read-fonts, but in write-fonts the extra table is completely superfluous, since in write-fonts we don't need to resolve offsets
- and so to improve the write-fonts API, we say, "this is actually just a vec, and here is how you convert it".

If this was a common pattern, it would make sense to support it in a general way, but it isn't (this is the only place we have encountered it so far). I don't think this is something we would ever want to infer in a schema, and I wouldn't really want to special-case it in the codegen. I think that if we were using a language agnostic schema we would probably just lose this annotation and accept a worse API.

Reply:
Problem is this moves us a step further away from a language-agnostic schema, it just got harder to get there in a meaningful way. IMHO it would be worth thinking through what it would take to not do this quite so Rust-ily.